docs/sdk-configuration.md

```diff
@@ -0,0 +1,676 @@
+
+```java
+class SleuthUsingService {
+  
+  @Inject
+  private OpenTelemetry openTelemetry;
+
+  public void doLogic() {
+    // My logic is important, so always sample it!
+    OpenTelemetrySdk.getTracerManagement().updateTraceConfig(config -> config.setSampler(ALWAYS_ON));
+    // This service was able to affect other services, even though Sleuth intends to
+    // "manage the SDK". Unlike the javaagent, it can't block access to SDK methods we may provide.
+    doSampledLogicWhileOtherServicesAlsoGetSampled();
+  }
+}
+```
+
+## TracerSdkManagement
+
+With the proposal to make a configured SDK fully built rather than allowing configuration
+mutability, the SDK management interface would only have `shutdown` and `forceFlush`. It seems
+reasonable to actually remove `shutdown` from the `TracerSdkProvider` - the SDK provider is mostly
+a bag of parts and has no intrinsic component, if instead components like `BatchSpanProcessor` and
+`SpanExporter` implement `Closeable`, we can model shutdown as the responsibility of the creator of
+the resources, which is idiomatic. In DI frameworks, this would generally be automatic. This would
+prevent the case where two `TracerProvider` share an exporter and one of them is shutdown - when
+shutting down the exporter, shut down the exporter.
+
+```java
+public class MyApp {
+ 
+    public static void main(String[] args) {
+        try (OtlpGrpcSpanExporter otlpExporter = OtlpGrpcSpanExporter.builder().build();
+              BatchSpanProcessor otlpProcessor = BatchSpanProcessor.builder(otlpExporter).build()) {
+            OpenTelemetrySdk sdk = OpenTelemetrySdk.builder()
+                .setTracerProvider(TracerSdkProvider.builder()
+                                        .addSpanProcessor(otlpProcessor)
+                                        .build())
+                .build();
+ 
+            Server server = Server.builder().setOpenTelemetry(sdk).build();
+            server.listenAndServe();
+        }
+    }
+
+}
+```
+
+```java
+@Component
+public class MyModule {
+
+    @Bean
+    // If SpanExporter extends Closeable, it's automatically closed
+    OtlpGrpcSpanExporter otlpExporter() {
+        return OtlpGrpcSpanExporter.builder().build()
+    }
+   
+    @Bean
+    // If BatchSpanProcessor extends Closeable, it's automatically closed
+    BatchSpanProcessor otlpSpanProcessor(OtlpGrpcSpanExporter otlpExporter) {
+        return BatchSpanProcessor.builder(otlpExporter).build();
+    }
+ 
+    @Bean
+    TracerSdkProvider tracerProvider(BatchSpanProcessor otlpSpanProcessor) {
+        return TracerSdkProvider.builder().addSpanProcessor(otlpSpanProcessor).build();
+    }
+
+    @Bean
+    OpenTelemetrySdk openTelemetry(TracerSdkProvider tracerProvider) {
+        return OpenTelemetrySdk.builder().setTracerProvider(tracerProvider).build();
+    }
+}
+```
+
+The `OpenTelemetrySdk` and `TracerSdkProvider` are still exposed to provide access to `forceFlush`.
```

Comment:
@jkwatson I realized it could make sense to remove `shutdown` from the `TracerSdkProvider`. It seems like it could make some lifecycle woes simpler to reason about, maybe even enable `toBuilder()`?